lfs.c

```diff
@@ -2730,14 +2730,18 @@ static int lfs_file_outline(lfs_t *lfs, lfs_file_t *file) {
 }
 #endif
 
-#ifndef LFS_READONLY
-static int lfs_file_flush(lfs_t *lfs, lfs_file_t *file) {
+static void lfs_file_invalidate_reading_flag(lfs_t *lfs, lfs_file_t *file) {
     if (file->flags & LFS_F_READING) {
         if (!(file->flags & LFS_F_INLINE)) {
             lfs_cache_drop(lfs, &file->cache);
         }
         file->flags &= ~LFS_F_READING;
     }
+}
+
+#ifndef LFS_READONLY
+static int lfs_file_flush(lfs_t *lfs, lfs_file_t *file) {
+    lfs_file_invalidate_reading_flag(lfs, file);
```

Comment:
Just a note, but I think this would be better structured by moving the `#ifndef LFS_READONLY` into `lfs_file_flush`. I will go ahead and do this after merging since I am hoping to make a patch release today.